### PR TITLE
Ext 946 dynamic page permissions

### PR DESCRIPTION
**Pull request by KevinMitchellREDBRICK**

### CODE REVIEW

File: src/jti_background.js [hunk 1]
```diff
@@ -445,19 +446,31 @@ function JSONtoTemplateData (templates) {
     return formattedTemplates;
 }
 
-function JSONtoDomainData (domains) {
-    var nextID = getNextID(domains);
+function JSONtoDomainData (domains, callback) {
+    var nextID = 1;
```
Comment (cfoucher) [inline src/jti_background.js]: Why hardcode this to one? What was wrong with getNextID?

File: src/jti_background.js [hunk 2]
```diff
@@ -468,6 +481,16 @@ function templateDataToJSON (templates) {
     return formattedTemplates;
 }
 
+function domainDataToJSON (domains) {
+    var formattedDomains = [];
+
+    $.each(domains, function (key, domain) {
+        delete domain.id;
```
Comment (cfoucher) [inline src/jti_background.js]: I don't think we actually need to delete the id. Aren't we simply creating a list of the `domain.name`'s?

File: src/jti_background.js (hunk 1 above)
Comment (KevinMitchellREDBRICK) [reply src/jti_background.js]: When JSONtoDomainData is called, we replace all the domain data. Therefore we start with a new set of IDs.

e.g. if I have 8 existing domains, then import from a file, in the old version the IDs would start at 9. Now they'll start at 1.

File: src/jti_background.js (hunk 2 above)
Comment (KevinMitchellREDBRICK) [reply src/jti_background.js]: :+1:

File: src/jti_background.js (hunk 1 above)
Comment (KevinMitchellREDBRICK) [reply src/jti_background.js]: nevermind, what I said above is nonsense. I'll dig into it.

Comment (KevinMitchellREDBRICK) [reply src/jti_background.js]: getNextID and =1 do the same thing here. getNextID() is heavier (needs a function call and some internal logic), but it's probably clearer what the intent of the program is, so I'll use it.

**Pull request merged**